internal/socket.go

```diff
@@ -70,9 +71,39 @@ func getUnixCreds(conn net.Conn) (*unix.Ucred, error) {
 		return nil, fmt.Errorf("doing rawConn Control: %w", err)
 	}
 
+	if err := authenticateUser(ucred); err != nil {
+		return nil, err
+	}
+
 	return ucred, nil
 }
 
+func authenticateUser(ucred *unix.Ucred) error {
```

Comment:
Could this be done by using syscalls instead of calling executables?

Reply:
Converted to syscalls what was possible, sudo check is cmd.Exec

Reply:
Converted all to syscalls